director/advertise.go

```diff
@@ -172,6 +173,17 @@ func AdvertiseOSDF() error {
 
 	for cacheAd, namespacesSlice := range cacheAdMap {
 		RecordAd(cacheAd, &namespacesSlice)
+
+		// In addition to recording the ads, we can add these to the string slice the Director uses
+		// when checking if specific hosts should receive a cache or an origin redirect. For now, when
+		// a cache uses the director as its redirector, it only makes sense to redirect the cache to an
+		// origin, regardless of what the director is configured to serve by default.
+		if param.Director_HostAwareRedirects.GetBool() {
```

Comment:
I'm not following this.

It appears it adds the hostname of each cache in the OSDF to the list of the hostnames the director identifies itself as.  Why is that?

It also adds it as a list, meaning this will grow every 10 minutes when this function is invoked.

Reply:
I think I got a bit confused as I was working on this, essentially mixing up the "from" and the "to". This feature would be nice to explore if the `Host` header contained the sender's host, which would allow us to say "requests **from** `Host: foo` should always be redirected to an origin," but the host header tells us **to**. I'll remove this chunk.